src/AWS.Deploy.Common/Recipes/Validation/OptionSettingItemValidators/SubnetsInVpcValidator.cs

```diff
@@ -39,15 +51,49 @@ public async Task<ValidationResult> Validate(object input, Recommendation recomm
             if (string.IsNullOrEmpty(vpcId))
                 return ValidationResult.Failed("The VpcId setting is not set or is empty. Make sure to set the VPC Id first.");
 
+            if (!string.IsNullOrEmpty(DefaultVpcOptionPath))
+            {
+                var useDefaultVpcSetting = _optionSettingHandler.GetOptionSetting(recommendation, DefaultVpcOptionPath);
+                var useDefaultVpcValue = _optionSettingHandler.GetOptionSettingValue<bool>(recommendation, useDefaultVpcSetting);
+
+                // For at least the ECS Fargate recipes, it's possible for the user to:
+                //   1. Select a non-default, existing VPC
+                //   2. Then set "Default VPC?" to true
+                // Because we have the default VPC selection take precedence in CDK projects over an explicit VPC Id,
+                // we want to validate the subnets against the default VPC in this case.
```

Comment:
The user can get in a weird state that looks like this:
![image](https://user-images.githubusercontent.com/1170880/198680341-0a4d239f-e9f4-4bc7-8e26-719e99d51293.png)

This block is my proposed fix, but it still requires the use to redo their selections.

I did explore setting the `VpcId` option to the default VPC automatically when the radio box is checked:
* I tried to do this in a new `Validator`, because I'm not aware of a precedent for options changing each others' settings based on some values.
* It would require a new network request from that validator to lookup the default VPC ID again.
* But it wouldn't update the Toolkit UI  - they'd still see the previously-selected VPC in the disabled box.

Reply:
I'm surprised the subnets even show up if `Default VPC` is selected. Should we update the recipe `Depends On` conditions to have the IsDefaultVPC = `false`.

Reply:
I don't really have a strong opinion. Both the Beanstalk and App Runner recipes allow you to select subnets even if the default VPC is selected.